### PR TITLE
Salesforce auth token to akka token

### PR DESCRIPTION
**Pull request by Calum-Campbell**
Previously the salesforce token was stored in a val that was obtained on startup, it then never refreshed itself. The token expires after 12 hours of inactivity so the app would be trying to use an expired token until it was redeployed and requested a fresh token. 

- Use a cache for the salesforce token. If it hasn't been used in 15 minutes request a fresh token.

### CODE REVIEW

File: app/services/SalesforceService.scala [hunk 1]
```diff
@@ -37,9 +39,12 @@ case class IdentityId(value: String, fieldName: String = "Zuora__Subscription__r
 case class Email(value: String, fieldName: String = "Zuora__Subscription__r.Zuora__CustomerAccount__r.Contact__r.Email") extends UniqueIdentifier
 case class SubscriptionId(value: String, fieldName: String = "Subscription_Name__c") extends UniqueIdentifier
 
-@Singleton class SalesforceService @Inject() (ws: WSClient)(implicit ec: ExecutionContext) extends Logging {
+@Singleton class SalesforceService @Inject() (ws: WSClient, actorSystem: ActorSystem)(implicit ec: ExecutionContext) extends Logging {
 
-  private lazy val sfAuth: SFAuthentication = {
+  val eventualAuthentication: Promise[Agent[SFAuthentication]] = Promise()
+  startAuth()
```
Comment (Calum-Campbell) [inline app/services/SalesforceService.scala]: I'm sure there is a better place to kick this off, not sure where though...

**Comment by rahilb** [comment]
using Guava's CacheBuilder: 

```
  val tokenCache = new CacheBuilder[NotUsed, String]()
    .maximumSize(1)
    .expireAfterWrite(15, TimeUnit.MINUTES)
    .build(new CacheLoader[NotUsed, String] {
      override def load(key: NotUsed): String = "salesforce-token"
    })
```
Then you can just replace all usages of the token val with `tokenCache.get(NotUsed)`

File: app/services/SalesforceService.scala [hunk 2]
```diff
@@ -37,33 +41,38 @@ case class IdentityId(value: String, fieldName: String = "Zuora__Subscription__r
...
   }
 
-  private val authHeader = ("Authorization", s"Bearer ${sfAuth.access_token}")
+  val sfAuthCache = CacheBuilder.newBuilder()
+    .maximumSize(1)
+    .expireAfterWrite(15, TimeUnit.MINUTES)
+    .build(cacheLoader)
+
+  val sfAuth = sfAuthCache.get(NotUsed)
+  val authHeader = ("Authorization", s"Bearer ${sfAuth.access_token}")
```
Comment (rahilb) [inline app/services/SalesforceService.scala]: Don't we want to make a new one every time?

**Comment by johnduffell** [comment]
good work, hope this will solve the issue! 🎉

**Pull request merged**